### PR TITLE
LCA of a Binary Tree

### LINKED ISSUE #7687: Lowest common ancestor of a binary tree
### Description

lowest common ancestor (lca) in a binary tree

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by stutimohta20**
<hr>

## Description 📜

Lowest Common Ancestor (LCA) of a node in a binary tree.
Fixes #7687 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by stutimohta20** [comment]
Please review my pr.

File: C_CPP/Data Structures/Trees/readme.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ## Trees
 
 - [Red Black Trees](RedBlackTrees)
-- [Flatten Binary Tree](FlattenBinaryTree.md)
\ No newline at end of file
+- [Flatten Binary Tree](FlattenBinaryTree.md)
+- [LCA of a Binary Tree](C_CPP/Data Structures/Trees/LCAofBinaryTree.md)
```
Comment (vasu-1) [inline C_CPP/Data Structures/Trees/readme.md:5]:
```suggestion
- [LCA of a Binary Tree](./LCAofBinaryTree.md)
```

File: C_CPP/Data Structures/Trees/LCAofBinaryTree.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# Lowest Common Ancestor of a Binary Tree
+
+## Binary Tree
+A Hierarchical data structure that stores data in the form of a tree and can have either **0,1 or 2 children** at each node.
+It works on the rule that a node which has lesser value that the root is place on the left subtree and the one with more value
```
Comment (vasu-1) [inline C_CPP/Data Structures/Trees/LCAofBinaryTree.md:5]:
```suggestion
It works on the rule that a node which has lesser value that the root is placed on the left subtree and the one with more value 
```

File: C_CPP/Data Structures/Trees/LCAofBinaryTree.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+- Pointer to right subtree.
+
+## Lowest Common Ancestor(LCA)
+The Lowest Common Ancestor (LCA) of two nodes in a rooted tree is the lowest (deepest) node that 
+is an ancestor of both the nodes.
+
+**Finding the LCA** - At first we traverse the tree and match nodes n1 and n2 if we get one of the 
+nodes in the left subtree and another one in the right subtree,then the root is the LCA.
+
+```
```
Comment (vasu-1) [inline C_CPP/Data Structures/Trees/LCAofBinaryTree.md:20]:
```suggestion
```cpp
```

for syntax effect

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Mention Time & Space Complexities !

**Comment by stutimohta20** [comment]
Made the changes.

**Comment by stutimohta20** [comment]
Please review @vasu-1

**Review by vasu-1 — APPROVED** [review]
LGTM @stutimohta20 

@Suvraneel

**Comment by Suvraneel** [comment]
@stutimohta20  Nice contribution 🚀  ❤️  :octocat: LGTM

**Pull request merged**